boards/st/nucleo_g431kb/nucleo_g431kb.dts

```diff
@@ -0,0 +1,119 @@
+/*
+ * Copyright (c) 2024 Klaus Nagel, <nagelkl01@gmail.com>
+ *
+ * SPDX-License-Identifier: Apache-2.0
+ */
+
+/dts-v1/;
+#include <st/g4/stm32g431Xb.dtsi>
+#include <st/g4/stm32g431k(6-8-b)tx-pinctrl.dtsi>
+
+/ {
+	model = "STMicroelectronics STM32G431KB-NUCLEO board";
+	compatible = "st,stm32g431kb-nucleo";
+
+	chosen {
+		zephyr,console = &stlink_serial;
+		zephyr,shell-uart = &stlink_serial;
+		zephyr,sram = &sram0;
+		zephyr,flash = &flash0;
+	};
+
+	leds: leds {
+		compatible = "gpio-leds";
+		green_led: led_0 {
+			gpios = <&gpiob 8 GPIO_ACTIVE_HIGH>;
+			label = "User LD2";
+		};
+
+	};
+
+	pwmleds {
+		compatible = "pwm-leds";
+
+		green_pwm_led: green_pwm_led {
+			pwms = <&pwm4 3 PWM_MSEC(20) PWM_POLARITY_NORMAL>;
+		};
+	};
+
+
+	aliases {
+		led0 = &green_led;
+		pwm-led0 = &green_pwm_led;
+		watchdog0 = &iwdg;
+	};
+};
+
+&clk_lsi {
+	status = "okay";
+};
+
+&clk_hsi {
+	status = "okay";
+};
+
+/* Adjust the pll for a SYSTEM Clock of 170MHz  */
+&pll {
+	div-m = <4>;
+	mul-n = <85>;
+	div-p = <7>;
+	div-q = <2>;
+	div-r = <2>;
+	clocks = <&clk_hsi>;
+	status = "okay";
+};
+
+&rcc {
+	clocks = <&pll>;
+	clock-frequency = <DT_FREQ_M(170)>;
+	ahb-prescaler = <1>;
+	apb1-prescaler = <1>;
+	apb2-prescaler = <1>;
+};
+
+&lpuart1 {
+	pinctrl-0 = <&lpuart1_tx_pa2 &lpuart1_rx_pa3>;
+	pinctrl-names = "default";
+	current-speed = <115200>;
+	status = "okay";
+};
+
+&timers4 {
+	st,prescaler = <10000>;
+	status = "okay";
+
+	pwm4: pwm {
+		status = "okay";
+		pinctrl-0 = <&tim4_ch3_pb8>;
+		pinctrl-names = "default";
+	};
+};
+
+stm32_lp_tick_source: &lptim1 {
+	clocks = <&rcc STM32_CLOCK_BUS_APB1 0x80000000>,
+			 <&rcc STM32_SRC_LSE LPTIM1_SEL(3)>;
+	status = "okay";
+};
```

Comment:
You're declaring LSE as LPTIM source clock but it is not enabled.
Do you actually need LPTIM (used for power mgmt).

Reply:
No, actually i don't neet power management, i will remove it

Reply:
When i delete the &lptim1 from the dts- file all tested samples build without error and run fine but all twister tests in tests/lib/cpp/cxx fail due to a warning:
_Kconfig for LPTIM source clock (LSE/LSE) is deprecated, use device tree_.

I found that in the final generated .config file there is the line 
_CONFIG_STM32_LPTIM_TIMER=y_
and
_CONFIG_STM32_LPTIM_CLOCK_LSI=y_

even if i delete the &lptim1 part from the dts- file.
I can not find where or how this Config Property gets set for the testsuite in tests/lib/cpp/cxx. 
Am I missing something here?

Reply:
The test case in question enables CONFIG_PM unconditionally, but on STM32 targets (from series which supports PM such as G4) a specific low power ticker configuration is required, which should be provided by lptim.

So, given this test configuration, you actually can't spare the following PM config:
```
&clk_lse {
	status = "okay";
};

stm32_lp_tick_source: &lptim1 {
	clocks = <&rcc STM32_CLOCK_BUS_APB1 0x80000000>,
		 <&rcc STM32_SRC_LSE LPTIM1_SEL(3)>;
	status = "okay";
};
```

Reply:
So for the PR i have to enable the lptim1 again in the Device Tree? Or should the test be changed? As i understand the purpose of the testcase is to check the different library configurations, why is there a dependency to power management?